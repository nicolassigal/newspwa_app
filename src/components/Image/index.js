import React, {PureComponent} from 'react';

class Image extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { intersected: false};
    this.observer = null;
  }

  componentDidMount() {
    this.observer = new IntersectionObserver(entries => {
      const image = entries[0];
      if (image.isIntersecting) {
        this.setState({ intersected: true });
        this.observer.disconnect();
      }
    });

    this.observer.observe(this.imgTag);
  }

  componentWillUnmount() {
    this.observer.disconnect();
  }

  render() {
    const { src, alt, ...props } = this.props;

    return (
        <img
        load="lazy"
        src={this.state.intersected ? src : '' }
        alt={alt || ''}
        ref={elem => (this.imgTag = elem)}
        {...props}/>
    );
  }
}

export default Image;
