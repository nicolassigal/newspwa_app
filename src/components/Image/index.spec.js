import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Image from './index';
import { render } from 'enzyme';

describe('Image component test', () => {
  it('Image component should render correctly', () => {
    const props = {
      alt: "alt",
      src: "src.png",
      onLoad: () => {}
    }


   const wrapper = render(<Image {...props}/>)
   expect(wrapper).toMatchSnapshot();

  });
});