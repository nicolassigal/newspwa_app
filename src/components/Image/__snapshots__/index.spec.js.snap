// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component test Image component should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "alt",
      "load": "lazy",
      "src": "",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "load": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "load": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
